config/confignet/confignet.go

```diff
@@ -5,10 +5,53 @@ package confignet // import "go.opentelemetry.io/collector/config/confignet"
 
 import (
 	"context"
+	"fmt"
 	"net"
 	"time"
 )
 
+// TransportType represents a type of network transport protocol
+type TransportType string
```

Comment:
If we go forward with this change, should we call this `TransportType` or `Transport`?  There was a comment [here](https://github.com/open-telemetry/opentelemetry-collector/issues/6490#issuecomment-1307663325) that maybe we shouldn't include `Type` in these names.

My opinion is to keep `Type` in the name.

Reply:
Related to https://github.com/open-telemetry/opentelemetry-collector/issues/9388

Reply:
I think we should finalize the discussion in https://github.com/open-telemetry/opentelemetry-collector/issues/9388 before making any changes here

Reply:
`Transport` sounds good to me in this particular case though

Reply:
Wrote up a quick summary of state of names here: https://github.com/open-telemetry/opentelemetry-collector/issues/9388#issuecomment-1915344957.  I think if we standardize on not including `Type` that there are some names in pdata that become "out-of-standard".  Since pdata is 1.0 and sets the precedence, I'm definitely in favor of keeping `Type` in the name.

I also personally find it clearer at a glance.  `confignet.Transport` conveys less information (in my opinion), than `confignet.TransportType`.  I feel like `Type` in the name quickly tells me it is used for enums.

Reply:
I think we should file a separate PR for the general rule and discuss it there (I mentioned what I think we should cover on https://github.com/open-telemetry/opentelemetry-collector/issues/9388#issuecomment-1919075284, if you think there are other items to add please say so!)

Reply:
Based on the new expectations for enumeration names, the name needs to convey a sense of categorization and I don't feel like the word `Transport` on its own is enough. I believe `TransportType` meets our expectations better.

Reply:
Hmm but that breaks user configurations, not just the API.  What do you think?

Reply:
Agreed, `TransportType` seems to fit the rules better